contracts/contracts/strategies/NativeStaking/ValidatorAccountant.sol

```diff
@@ -0,0 +1,257 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { Pausable } from "@openzeppelin/contracts/security/Pausable.sol";
+import { ValidatorRegistrator } from "./ValidatorRegistrator.sol";
+import { IWETH9 } from "../../interfaces/IWETH9.sol";
+
+/// @title Accountant of the rewards Beacon Chain ETH
+/// @notice This contract contains the logic to attribute the Beacon Chain swept ETH either to full
+/// or partial withdrawals
+/// @author Origin Protocol Inc
+abstract contract ValidatorAccountant is ValidatorRegistrator {
+    address public immutable VAULT_ADDRESS;
+
+    /// @dev The WETH present on this contract will come from 2 sources:
+    ///  - as a result of deposits from the VaultAdmin
+    ///  - accounting function converting beaconChain rewards from ETH to WETH
+    ///
+    /// We need to be able to keep a separate accounting of the WETH so we understand how much we can pass oh to
+    /// the harvester as a consequence of rewards harvesting and how much registrator can pick up as a result of WETH
+    /// deposit into the strategy contract.
+    /// To achieve this the beacon chain rewards are accounted for using below variable, all other WETH is assumed to be
+    /// present as a result of a deposit.
+    uint256 public beaconChainRewardWETH = 0;
+
+    /// @dev start of fuse interval
+    uint256 public fuseIntervalStart = 0;
+    /// @dev end of fuse interval
+    uint256 public fuseIntervalEnd = 0;
+    /// @dev Governor that can manually correct the accounting
+    address public accountingGovernor;
+    /// @dev Strategist that can pause the accounting
+    address public strategist;
```

Comment:
Best to read the `strategist` from the Vault rather than duplicating the maintenance in the strategy. This is what OETH AMO strategy does

```Solidity
    modifier onlyStrategist() {
        require(
            msg.sender == IVault(vaultAddress).strategistAddr(),
            "Caller is not the Strategist"
        );
        _;
    }
```